sys-devel/flang/flang-13.0.0.ebuild

```
@@ -0,0 +1,100 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CMAKE_ECLASS=cmake
+
+PYTHON_COMPAT=( python3_{8..10} )
+inherit cmake-multilib llvm python-single-r1
+
+DESCRIPTION="Flang is a ground-up implementation of a Fortran front end written in modern C++"
+## In order to compile this package you need sys-devel/mlir package
+HOMEPAGE=" https://github.com/llvm/llvm-project/tree/main/flang"
+SRC_URI=" https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/flang-13.0.0.src.tar.xz -> ${P}.tar.xz"
+
+LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA MIT"
+SLOT="13"
+IUSE="doc test +system-llvm"
+KEYWORDS="~amd64"
+
+DEPEND="
+	>=sys-devel/clang-13.0.0:13=[static-analyzer,${MULTILIB_USEDEP}]
+	sys-devel/llvm:13=[${MULTILIB_USEDEP}]
+"
+RDEPEND="${DEPEND}
+	${PYTHON_DEPS}"
+
+BDEPEND="
+	doc? (
+		dev-python/sphinx
+		app-doc/doxygen[dot]
+	)
+	test? (
+		dev-python/lit
+	)
+"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+LLVM_MAX_SLOT=13
+
+src_unpack() {
+	unpack ${P}.tar.xz
+	mv "${WORKDIR}/${P}.src" "${WORKDIR}/${P}"
+}
+
+PATCHES=(
+	"${FILESDIR}"/flang-Link-against-libclang-cpp.so.patch
+	"${FILESDIR}"/flang-PATCH-flang-Disable-use-of-sphinx_markdown_tables.patch
+)
+
+pkg_setup() {
+	use system-llvm && llvm_pkg_setup
+	use system-llvm &&  export CC=clang
```

Don't align:
```suggestion
	use system-llvm && export CC=clang
```

but this isn't ideal anyway, it doesn't respect the user's choice for these variables. Also, if we _were_ going to  do this, we should do:
`if use system-llvm ; then
	export ...
fi`

I think you should be able to just get rid of this, right? We always want to use the system LLVM copy.

Or must Flang be built with Clang specifically or something (which will require some machinery but not quite like this)?

---

The problem is as follows. 
flang could be compiled either by g++ or by clang+.  Because gentoo is about choices, I would like to leave the possibility of choosing g++/clang+ - maybe some other USE flag could be chosen ? 
Is there any USE flag which determines CC/CXX compiler for particular ebuild despite of system wide CC/CXX compiler ?
And one more think, it is enough to set up CC and CXX for flang. 

Personally, on my computer I stick with GNU toolchain (particularly CXX=g++), but when I choose clang+ for flang   all warning disappear.